core/streamdata.go

```diff
@@ -42,88 +43,47 @@ func (md *SegTranscodingMetadata) Flatten() []byte {
 	return buf
 }
 
-func RandomVideoID() []byte {
-	return RandomIdGenerator(HashLength)
-}
-
-func RandomIdGenerator(length uint) []byte {
+var RandomIdGenerator = func(length uint) []byte {
 	x := make([]byte, length, length)
 	for i := 0; i < len(x); i++ {
 		x[i] = byte(rand.Uint32())
 	}
 	return x
 }
 
-//StreamID is VideoID|Rendition
-type StreamID string
-
-func MakeStreamID(id []byte, rendition string) (StreamID, error) {
-	if len(id) != HashLength || rendition == "" {
-		return "", ErrStreamID
-	}
-	return StreamID(fmt.Sprintf("%x%v", id, rendition)), nil
-}
+type ManifestID string
```

Comment:
> What did you have in mind?

I'm actually not sure if it's possible in golang with primitive types; the goal would be to prevent type casting entirely. It It'd be similar in spirit to hiding non-exported opaque structs behind a public interface. If we used an empty interface here, I think we'd also lose the ability to use the type as keys in maps.

```
foo.h:
struct foo_t;

foo.c:
struct foo_t {
  int a;
};
```

Here, only `foo.c` has access to the definition of foo_t. All other files see it as a opaque type of `struct foo_t` with a hidden definition. Even in C, this doesn't work for primitive types, but it is possible to make opaque primitive types with OCaml modules:

```
foo.mli:
type foo

foo.ml:
type foo = string
```

Reply:
Left the ManifestID type as-is for now, and the Rendition is still a string. We can revisit later.